aiida/tools/importexport/dbexport/__init__.py

```diff
@@ -266,15 +253,19 @@ def export(
 
         # Create a mapping of node PK to UUID.
         node_pk_2_uuid_mapping: Dict[int, str] = {}
+        repository_metadata_mapping: Dict[int, dict] = {}
+
         if node_ids_to_be_exported:
             qbuilder = orm.QueryBuilder().append(
                 orm.Node,
-                project=('id', 'uuid'),
+                project=('id', 'uuid', 'repository_metadata'),
                 filters={'id': {
                     'in': node_ids_to_be_exported
                 }},
             )
-            node_pk_2_uuid_mapping = dict(qbuilder.all(batch_size=batch_size))
+            for pk, uuid, repository_metadata in qbuilder.iterall(batch_size=batch_size):
+                node_pk_2_uuid_mapping[pk] = uuid
+                repository_metadata_mapping[pk] = repository_metadata
```

Comment:
note this will increase the memory overhead.
We should consider whether we want to do it this way (quicker but more memory) or obtain this data in a separate iterable query in `_write_node_repositories` (slower but less memory), e.g.

```python
            qbuilder = orm.QueryBuilder().append(
                orm.Node,
                project=('repository_metadata',),
                filters={'id': {
                    'in': node_ids_to_be_exported
                }},
            )
            for repository_metadata, in qbuilder.iterall(batch_size=batch_size):
                collect_hashkeys(repository_metadata.get('o', {}))
```

Reply:
I agree, but before changing this, we should do some benchmarking to see if the added memory usage may become problematic